content/200-concepts/100-components/02-prisma-client/260-client-extensions/140-shared-extensions.mdx

```diff
@@ -0,0 +1,93 @@
+---
+title: 'Shared Prisma Client extensions'
+metaTitle: 'Shared Prisma Client extensions (Preview)'
+metaDescription: 'Share extensions or import shared extensions into your Prisma project'
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+You can share your [Prisma Client extensions](/concepts/components/prisma-client/client-extensions/) with other users, either as packages or as code snippets, and import extensions that other users create into your Prisma projects.
+
+</TopBlock>
+
+## Share Prisma Client extensions
+
+To create an extension that is more likely to work for other users, use the `$all` prefix, for example [`$allModels`](/concepts/components/prisma-client/client-extensions/model#add-a-custom-method-to-all-models-in-your-schema). This extension applies generally, not just to your specific schema.
+
+For example, a generic extension might take the following form:
+
+```ts
+const xprisma = new prisma.$extends({
+  model: {
+    $allModels: {
+      getClass<T extends object>(this: T): new () => T {
+        return class {
+          /** Generic Implementation */
+        } as any
+      },
+    },
+  },
+})
+
+class UserService extends xprisma.user.getClass() {
+  method() {
+    const user = this.findFirst({}) // fully type-safe
```

Comment:
What is `this` here? Is that 100% obvious to someone that knows JS/TS?

Reply:
I've added a note to the draft that we need to clarify this example or select a different example